lib/apiservers/portlayer/restapi/handlers/storage_handlers.go

```diff
@@ -68,32 +68,27 @@ func (h *StorageHandlersImpl) Configure(api *operations.PortLayerAPI, handlerCtx
 	// expensive metadata lookups.
 	h.imageCache = spl.NewLookupCache(ds)
 
-	// The same is done for volumes.  It's implemented via a cache which writes
-	// to an implementation that takes a datastore to write to.
-	vsVolumeStore, err := vsphereSpl.NewVolumeStore(op, handlerCtx.Session)
-	if err != nil {
-		log.Panicf("Cannot instantiate the volume store: %s", err)
-	}
-
 	// Get the datastores for volumes.
 	// Each volume store name maps to a datastore + path, which can be referred to by the name.
 	dstores, err := datastore.GetDatastores(op, handlerCtx.Session, spl.Config.VolumeLocations)
 	if err != nil {
 		log.Panicf("Cannot find datastores: %s", err)
 	}
 
+	h.volumeCache = spl.NewVolumeLookupCache(op)
+
 	// Add datastores to the vsphere volume store impl
 	for volStoreName, volDatastore := range dstores {
 		log.Infof("Adding volume store %s (%s)", volStoreName, volDatastore.RootURL)
-		_, err := vsVolumeStore.AddStore(op, volDatastore, volStoreName)
+
+		vs, err := vsphere.NewVolumeStore(op, volStoreName, handlerCtx.Session, volDatastore)
 		if err != nil {
-			log.Errorf("volume addition error %s", err)
+			log.Panicf("Cannot instantiate the volume store: %s", err)
```

Comment:
If we were to retry, it would be in the called function (`NewVolumeStore`).  

That said, I agree, `Panic`ing isn't the right thing to do here.  Will lower the severity to `Errorf`.